Add ES5+CommonJS support

beck opened the pull request:
Compile all the source at prepublish.

Motivation here is to test a consumer using node.

ATM, a consumer's test will fail on the ESM syntax (which we workaround by
browswerifying, making unit tests very slow).

The cost to adding CommonJS support is that the module can only have
one export and is why the member exports have been removed.

beck committed:
Add ES5+CommonJS support

Compile all the source at prepublish.

Motivation here is to test a consumer using node.

ATM, a consumer's test will fail on the ESM syntax (which we workaround by
browswerifying, making unit tests very slow).

The cost to adding CommonJS support is that the module can only have
one export and is why the member exports have been removed.

snitch commented:
:sparkles: No lint errors found. :sparkles:

stefanor commented on scripts/compile-common-js.sh:
```diff
@@ -0,0 +1,23 @@
+echo "Compiling to ES5+CommonJS ..."
+for f in $(find src -name "*.js")
+do
+    if [[ $f == *"-spec.js"* ]]; then
```
Considered adding `! -name '*-spec.js'` to the `find`?

beck committed:
Add an npmignore to override gitignore

Otherwise compiled js is not included in the release.

beck committed:
Cleanup bash

snitch commented:
:sparkles: No lint errors found. :sparkles:

beck committed:
Cleanup published artifact

Use `.npmignore` to exclude tests and dev configs.

Using `npm pack` the package is now:
```
package
├── API.md
├── CHANGELOG.md
├── LICENSE
├── README.md
├── cjs
│   ├── actions.js
│   ├── helpers
│   │   ├── query-handler-stub.js
│   │   └── request-spy.js
│   ├── index.js
│   ├── page-merger.js
│   ├── paginator-error.js
│   ├── paginators
│   │   ├── base-paginator.js
│   │   ├── limit-offset-paginator.js
│   │   └── page-number-paginator.js
│   ├── query-handlers
│   │   ├── limit-offset-query-handler.js
│   │   └── page-number-query-handler.js
│   └── test-setup.js
├── package.json
└── src
    ├── actions.js
    ├── fixtures
    │   └── page-number-response.json
    ├── helpers
    │   ├── query-handler-stub.js
    │   └── request-spy.js
    ├── index.js
    ├── page-merger.js
    ├── paginator-error.js
    ├── paginators
    │   ├── base-paginator.js
    │   ├── limit-offset-paginator.js
    │   └── page-number-paginator.js
    └── query-handlers
        ├── limit-offset-query-handler.js
        └── page-number-query-handler.js
```

snitch commented:
:sparkles: No lint errors found. :sparkles:

stefanor commented on scripts/compile-common-js.sh:
```diff
@@ -0,0 +1,20 @@
+rm -rf cjs
+
+echo "Compiling to ES5+CommonJS ..."
+for f in $(find src -name "*.js" ! -name '*-spec.js')
```
Consistent quotes?

beck replied:
Shame on me for copy/pasting your critique

stefanor replied:
Yeah, I realised that I'm to blame for that :P

beck committed:
Cleanup quotes

snitch commented:
:sparkles: No lint errors found. :sparkles:
